scripts/west_commands/sign.py

```diff
@@ -18,8 +18,7 @@
     FIND_BUILD_DIR_DESCRIPTION
 from runners.core import BuildConfiguration
 from zcmake import CMakeCache
-from zephyr_ext_common import Forceable, load_dot_config, \
-    ZEPHYR_SCRIPTS
+from zephyr_ext_common import Forceable, ZEPHYR_SCRIPTS
```

Comment:
Ah, I seem to have forgotten about BuildConfiguration when I wrote `load_dot_config()`.

If you check the implementation of `load_dot_config()`, it is actually more robust than `BuildConfiguration._parse`. For example, `BuildConfiguration` does not understand that `# CONFIG_FOO is not set` really means `CONFIG_FOO=n`.

So making this change is a bit of a step backwards in terms of compatibility with kconfiglib.

I suggest that what we do is move the code for `load_dot_config` into `BuildConfiguration._parse`, since `load_dot_config` is actually doing the right thing (even for substrings).

Reply:
> 
> If you check the implementation of `load_dot_config()`, it is actually more robust than `BuildConfiguration._parse`. For example, `BuildConfiguration` does not understand that `# CONFIG_FOO is not set` really means `CONFIG_FOO=n`.
> 
Yes I know I have seen this, but the `BuildConfiguration` is used more often...

> So making this change is a bit of a step backwards in terms of compatibility with kconfiglib.
> 
> I suggest that what we do is move the code for `load_dot_config` into `BuildConfiguration._parse`, since `load_dot_config` is actually doing the right thing (even for substrings).

.. and yes I was going to propose changing that next, though as it does not seem to be that crucial: the `get(key, alternative_value)` fills the gap

Reply:
> Yes I know I have seen this, but the `BuildConfiguration` is used more often...

Right, that's why I think we can just move load_dot_config's code into BuildConfiguration's _parse method, right?

Reply:
OK, I will do that.

Reply:
Ok, so I have partially merged the `load_dot_config` into the _parse().
I have left out generating dictionary entries for the "unset values", for following reasons:
1) The `n` is not the same as unset because unset value may be also number or string, which means that in case where string or number is unset it being actually set to 'n';
2) not adding option to directory allows to detect that the option is really unset rather set to `n`
3) The default value parameter to `get` works better than defaulting all unset options to `n` because user does know what is the type of an option that is being checked with `get` and can decide what should the default be, or can detect if value is unset;
4) It is not true that .config is not always automatically generated and contains strict syntax: for example I tend to edit parameters directly in .config; this means that some lines get commented out by me and are not ending wit "not set" and fail to be properly matched.